dbt/include/global_project/macros/materializations/models/incremental/merge.sql

```diff
@@ -61,34 +61,23 @@
     {%- set dest_cols_csv = get_quoted_csv(dest_columns | map(attribute="name")) -%}
 
     {% if unique_key %}
-        {% if unique_key is sequence and unique_key is not string %}
-            delete from {{target }}
-            using {{ source }}
-            where (
-                {% for key in unique_key %}
-                    {{ source }}.{{ key }} = {{ target }}.{{ key }}
```

Comment:
@peterallenwebb and I took a look at this PR today.

This line of code is updated by https://github.com/dbt-labs/dbt-adapters/pull/110, so we think that PR should be reviewed/merged prior to reviewing this PR further.